frame/support/procedural/src/transactional.rs

```diff
@@ -49,7 +49,9 @@ pub fn require_transactional(_attr: TokenStream, input: TokenStream) -> Result<T
 	let output = quote! {
 		#(#attrs)*
 		#vis #sig {
-			#crate_::storage::require_transaction();
+			if !#crate_::storage::is_transactional() {
+				return Err(#crate_::pallet_prelude::DispatchError::TransactionalLimit.into());
```

Comment:
I think we should have a separate error here for "Layer expected". "Too Many layers" is quite a different error.